src/amo/reducers/addonsByAuthors.js

```diff
@@ -78,32 +83,86 @@ type LoadAddonsByAuthorsAction = {|
 |};
 
 export const loadAddonsByAuthors = (
-  { addons, forAddonSlug }: LoadAddonsByAuthorsParams
+  { addons, addonType, authors, forAddonSlug }: LoadAddonsByAuthorsParams
 ): LoadAddonsByAuthorsAction => {
   invariant(addons, 'A set of add-ons is required.');
+  invariant(authors, 'A list of authors is required.');
 
   return {
     type: LOAD_ADDONS_BY_AUTHORS,
-    payload: { addons, forAddonSlug },
+    payload: { addons, addonType, authors, forAddonSlug },
   };
 };
 
-export const getAddonsForSlug = (state: State, slug: string) => {
+export const joinAuthorNames = (
+  authorNames: Array<string>, addonType?: string
+) => {
+  return authorNames.sort().join('-') + (addonType ? `-${addonType}` : '');
+};
+
+export const getLoadingForAuthorNames = (
+  state: AddonsByAuthors, authorNames: Array<string>, addonType?: string
+) => {
+  return authorNames && authorNames.length ?
+    (state.loadingFor[joinAuthorNames(authorNames, addonType)] || null) : null;
+};
+
+export const getAddonsForSlug = (
+  state: AddonsByAuthors,
+  slug: string,
+) => {
   const ids = state.byAddonSlug[slug];
 
   return ids ? ids.map((id) => {
     return state.byAddonId[id];
   }) : null;
 };
 
+export const getAddonsForUsernames = (
+  state: AddonsByAuthors,
+  usernames: Array<string>,
+  addonType?: string,
+  excludeSlug?: string,
+) => {
+  invariant(usernames && usernames.length, 'At least one username is required');
+
+  const ids = usernames.map((username) => {
+    return state.byUsername[username];
+  }).reduce((array, addonIds) => {
+    if (!addonIds || !array) {
+      return null;
+    }
+
+    for (const addonId of addonIds) {
```

Comment:
Probably yeah, but I'm not sure what it buys us for this? I don't think we use `Set` elsewhere in the code (though I think we did on the linter) fwiw

Reply:
Oh wait, right, because I wouldn't have to check using `includes`.

I'll give it a go; if flow doesn't mind because it ends up throwing around a different type I'm for it. 👍

Reply:
Yes, it was to avoid the `includes` check. In fact, we could avoid a few more `includes` checks if we converted the stored objects e.g., `byAddonSlug` and `byUserId` to store a `Set` instead of an `Array`, but your comment below suggests that it's not as simple as it sounds.

It was just a suggestion and I'm fine with leaving it as is for now.